### ISSUE TITLE
CTE With Partition OVER Example

### ISSUE THREAD

**Issue by vertonghenb** [HOGENT-Databases/DB2-Workshops#1]
Introduce a rather compex but usefull Recursive CTE with a Partition OVER
```sql
WITH cte_product_uses_component AS (
    SELECT
    Id = Product.Id,
    [Path] = CAST(Component.Code AS NVARCHAR(MAX)),
    Product.Code,
    Depth = 1
    FROM Product
            JOIN Formula on Formula.ProductId = Product.Id
            JOIN FormulaLine ON FormulaLine.FormulaId = Formula.Id
                JOIN Product Component ON FormulaLine.ComponentId = Component.Id
    WHERE Component.Code = 'SO.206.001' AND Product.Enabled = 1
    UNION ALL
    -- GET ALL PRODUCTS WHERE COMPONENTID == Product.ID
    SELECT
    Id = Product.Id,
    [Path] = [Path]  + ' - ' + Component.Code,
    Product.Code,
    Depth = cte_product_uses_component.Depth + 1
    FROM Product
            JOIN Formula on Formula.ProductId = Product.Id
            JOIN FormulaLine ON FormulaLine.FormulaId = Formula.Id
                JOIN Product Component ON FormulaLine.ComponentId = Component.Id
                    JOIN cte_product_uses_component  ON cte_product_uses_component.Id = Component.Id AND Product.Enabled = 1
)

-- Basically we're doing a DISTINCT ON (ID) here.
-- Each Id get's his own RowNumber, and we're only selecting the first one.
SELECT  *
FROM    (SELECT ID, Depth,[Path], Code,
                ROW_NUMBER() OVER (PARTITION BY Id ORDER BY Id) AS RowNumber
         FROM   cte_product_uses_component
         ) AS partitionedById
WHERE   partitionedById.RowNumber = 1 -- Only select the first row
ORDER BY Depth desc
```
